cdap-docs/cdap-apps/source/hydrator/custom.rst

```diff
@@ -449,3 +449,255 @@ Plugin Deployment
 .. include:: ../../../developers-manual/source/building-blocks/plugins.rst 
    :start-after: .. _plugins-deployment-artifact:
    :end-before:  .. _plugins-use-case:
+
+Configuration JSON for a plugin (for Webapp UI alone)
+======================================================
+Every property of a plugin is represented, by default, as input fields in the UI. This can be however changed to specific widgets that closely match the type of the property in a plugin. This can be specified as JSON for the UI to consume.
+
+The configuration JSON is composed of a list of property groups and a list of outputs. For instance the JSON would look something like this,
+
+.. code-block:: Javascript
+
+  {
+    "configuration-groups": [
+      "group1",
+      "group2",
+      ...
+    ],
+    "outputs": [
+      {"ouput-property-1"},
+      {"ouput-property-2"},
+      ...
+    ]
+  }
+
+Configuration groups
+-------------------
+  Configuration groups are simple grouping of properties in a plugin. For instance in a Stream Source plugin - Stream Name, Duration & Delay could be grouped as Stream Configuration. So based on this example a configuration group can be represented as an object with a name & a list of properties of the plugin that falls under that group. In the case of Stream source plugin it would look something like this,
+
+  .. code-block:: Javascript
+
+    {
+      "configuration-groups": [
+        {
+          "name": "Stream Configuration",
+          "properties": [
+            {"field1"},
+            {"field2"},
+            {"field3"}
+          ]
+        }
+      ],
+      outputs: [
+        {output-property1},
+        {output-property2},
+        ..
+      ]
+    }
+  Once a group is established we can configure how each field inside the group is represented in the UI. The configuration of each property of the plugin is composed of following parts,
```

Comment:
I think this line should not end with a comma. Rather, it should end with something like a colon.